overwatch-rs/src/services/relay.rs

```diff
@@ -141,13 +141,13 @@ impl<S: ServiceCore> Relay<S> {
     }
 
     #[instrument(skip(self), err(Debug))]
-    pub async fn connect(&mut self) -> Result<OutboundRelay<S::Message>, RelayError> {
```

Comment:
btw I think it was my bad, this could have been a `self` function from the start

Reply:
Actually yes, this should consume the relay probably.